src/unit_tests/wazuh_modules/scheduling/wmodules_scheduling_helpers.c

```diff
@@ -117,7 +117,11 @@ void check_day_of_week(const sched_scan_config *scan_config, struct tm *date_arr
         assert_int_equal( scan_config->scan_wday, date_array[i].tm_wday);
         if(i > 0){
             // Assert there is one week difference
-            assert_int_equal((date_array[i-1].tm_yday + 7) % 365, date_array[i].tm_yday);
+            if (date_array[i-1].tm_year % 4 == 0) { //leap year
```

Comment:
Not every year divisible by 4 is leap.

> Every year that is exactly divisible by four is a leap year, except for years that are exactly divisible by 100, but these centurial years are leap years if they are exactly divisible by 400. For example, the years 1700, 1800, and 1900 are not leap years, but the years 1600 and 2000 are.

(https://en.wikipedia.org/wiki/Gregorian_calendar)